pinot-common/src/main/java/com/linkedin/pinot/common/data/FieldSpec.java

```diff
@@ -207,57 +184,73 @@ public Object getDefaultNullValue() {
             return DEFAULT_DIM_NULL_VALUE_OF_FLOAT;
           case DOUBLE:
             return DEFAULT_DIM_NULL_VALUE_OF_DOUBLE;
+          case BOOLEAN:
+          case CHAR:
           case STRING:
             return DEFAULT_DIM_NULL_VALUE_OF_STRING;
           default:
             throw new UnsupportedOperationException(
                 "Unknown default null value for dimension/time column of data type " + _dataType);
         }
       default:
-        throw new UnsupportedOperationException("Unknown field type" + _fieldType);
+        throw new UnsupportedOperationException("Unsupported field type" + _fieldType);
     }
   }
 
   @Override
   public String toString() {
-    Object defaultNullValue = new String("N/A");
-    try {
-      defaultNullValue = getDefaultNullValue();
-    } catch (UnsupportedOperationException e) {
-      // Ignore the exception
-    }
-    return "< data type: " + getDataType() + " , field type: " + getFieldType()
-        + (isSingleValueField() ? ", single value column" : ", multi value column, delimiter: '" + getDelimiter() + "'")
-        + ", default null value: " + defaultNullValue + " >";
+    return "< field name: " + _name + ", field type: " + _fieldType + ", data type: " + _dataType
+        + (_isSingleValueField ? ", single-value field" : ", multi-value field, delimiter: '" + _delimiter + "'")
+        + ", default null value: " + getDefaultNullValue() + " >";
   }
 
   @Override
-  public boolean equals(Object obj) {
-    if (obj instanceof FieldSpec) {
-      return toString().equals(obj.toString());
+  public boolean equals(Object anObject) {
+    if (this == anObject) {
+      return true;
+    }
+    if (anObject instanceof FieldSpec) {
```

Comment:
Compare type properly so that a child class of this one should not compare as equals.